ipaserver/plugins/certmap.py

```diff
@@ -0,0 +1,355 @@
+        ),
+        Str(
+            'ipacertmapmatchrule?',
+            cli_name='matchrule',
+            label=_('Matching rule'),
+            doc=_('Rule used to check if a certificate can be used for'
+                  ' authentication'),
+        ),
+        DNSNameParam(
+            'associateddomain*',
+            cli_name='domain',
+            label=_('Domain name'),
+            doc=_('Domain where the user entry will be searched'),
+        ),
+        Int(
+            'ipacertmappriority?',
+            cli_name='priority',
+            label=_('Priority'),
+            doc=_('Priority of the rule (higher number means lower priority'),
+            minvalue=0,
+        ),
+        Flag(
+            'ipaenabledflag?',
+            label=_('Enabled'),
+            flags=['no_option'],
+            default=True
+        ),
+    )
+
+    permission_filter_objectclasses = ['ipacertmaprule']
+    managed_permissions = {
+        'System: Add Certmap Rules': {
+            'replaces_global_anonymous_aci': True,
+            'ipapermright': {'add'},
+            'default_privileges': {
+                'Certificate Identity Mapping Administrators'},
+        },
+        'System: Read Certmap Rules': {
+            'replaces_global_anonymous_aci': True,
+            'ipapermbindruletype': 'all',
+            'ipapermright': {'read', 'search', 'compare'},
+            'ipapermdefaultattr': {
+                'objectclass', 'cn', 'description',
+                'ipacertmapmaprule', 'ipacertmapmatchrule', 'associateddomain',
+                'ipacertmappriority', 'ipaenabledflag',
+            },
+        },
+        'System: Delete Certmap Rules': {
+            'replaces_global_anonymous_aci': True,
+            'ipapermright': {'delete'},
+            'default_privileges': {
+                'Certificate Identity Mapping Administrators'},
+        },
+        'System: Modify Certmap Rules': {
+            'replaces_global_anonymous_aci': True,
+            'ipapermright': {'write'},
+            'ipapermdefaultattr': {
+                'objectclass', 'cn', 'description',
+                'ipacertmapmaprule', 'ipacertmapmatchrule', 'associateddomain',
+                'ipacertmappriority', 'ipaenabledflag',
+            },
+            'default_privileges': {
+                'Certificate Identity Mapping Administrators'},
+        },
+    }
+
+    def get_dn(self, *keys, **options):
+        rulename = keys[-1]
+        dn = super(certmaprule, self).get_dn(rulename, **options)
+        return dn
+
+
+@register()
+class certmaprule_add(LDAPCreate):
+    __doc__ = _('Create a new Certificate Identity Mapping Rule.')
+
+    msg_summary = _('Added Certificate Identity Mapping Rule "%(value)s"')
+
+
+@register()
+class certmaprule_mod(LDAPUpdate):
+    __doc__ = _('Modify a Certificate Identity Mapping Rule.')
+
+    msg_summary = _('Modified Certificate Identity Mapping Rule "%(value)s"')
+
```

Comment:
@HonzaCholasta 
I discussed this point offline with Sumit and his answer was that any domain name should be accepted. If SSSD cannot make sense of one of the domains it will ignore it and log a message